cmd/ttn-lw-cli/commands/end_devices_split.go

```diff
@@ -34,20 +35,36 @@ var (
 	setEndDeviceToJS   = ttnpb.AllowedFieldMaskPathsForRPC["/ttn.lorawan.v3.JsEndDeviceRegistry/Set"]
 )
 
+func nonImplicitPaths(paths ...string) []string {
+	nonImplicitPaths := make([]string, 0, len(paths))
+	for _, path := range paths {
+		if path == "ids" || strings.HasPrefix(path, "ids.") {
+			continue
+		}
+		if path == "created_at" || path == "updated_at" {
```

Comment:
Why would these be implicit on `Get`?
On `Set` it makes sense - the entity is modified.
On `Get` there is no reason to return something user does not ask for.

Reply:
In fact, we may want to allow users to `set` the `created_at` in case of, e.g. migration.
Leave this logic to the registries.

Reply:
1. `created_at` and `updated_at` are modified by the server. I don't think it would be a good idea to give the possibility/responsibility of setting those to the client.
2. The `created_at` and `updated_at` fields are non-nullable in our proto definitions, so if we don't fetch them, users will always see a zero date printed out, which isn't what we want. Therefore I think they should be implicitly part of the field mask.
3. This logic filters out these implicit fields so that we won't call the NS only to fetch the `ids`, `created_at` and `updated_at` (the CLI determines which servers to call based on the field paths that it's allowed to GET from them)

@johanstokking can you also share your opinions here?

Reply:
Agreed, but we may want to come back on setting the `created_at` and `updated_at` fields later. But that's out of scope here.